kernel/declarations.mli

```diff
@@ -120,14 +120,11 @@ type 'opaque pconstant_body = {
 type constant_body = Opaqueproof.opaque pconstant_body
 
 (** {6 Representation of mutual inductive types in the kernel } *)
-type nested_type =
-| NestedInd of inductive
-| NestedPrimitive of Constant.t
 
 type recarg =
 | Norec
 | Mrec of inductive
-| Nested of nested_type
+| NestedPrimitive of Constant.t
```

Comment:
I feel like it could instead make sense to have 
~~~ocaml
type nested_type =
| NestedInd of inductive
| NestedPrimitive of Constant.t

type recarg =
| Norec
| Mrec of nested_type
~~~

Reply:
You mean so that, e.g., we can enforce statically that `mk_paths` does not take Norec as an argument? Why not, but I would then remove "nested" in the name, because it is not necessarily nested. So, something like:
```coq
type recarg_type =
| RecArgInd of inductive
| RecArgPrim of constant

type recarg =
| Norec
| Mrec of recarg_type
```

Reply:
>You mean so that, e.g., we can enforce statically that mk_paths does not take Norec as an argument?

I just think it looks more uniform to treat inductive and primitive in the same way

Reply:
So renaming Mrec into MrecInd and NestedPrimitive into MrecPrim would also be ok?

Reply:
No, that doesn't factor them